products/bigquery/api.yaml

```diff
@@ -1433,3 +1433,139 @@ objects:
                         - :INTEGER
                         - :FLOAT
                         - :BOOLEAN
+  - !ruby/object:Api::Resource
+    name: 'Routine'
+    kind: 'bigquery#routine'
+    base_url: projects/{{project}}/datasets/{{dataset_id}}/routines
+    self_link: projects/{{project}}/datasets/{{dataset_id}}/routines/{{routine_id}}
+    description: |
+      A user-defined function or a stored procedure that belongs to a Dataset
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Routines Intro': 'https://cloud.google.com/bigquery/docs/reference/rest/v2/routines'
+      api: 'https://cloud.google.com/bigquery/docs/reference/rest/v2/routines'
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: routineReference
+        description: Reference describing the ID of this routine
+        required: true
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'datasetId'
+            description: The ID of the dataset containing this routine
+            required: true
+          - !ruby/object:Api::Type::String
+            name: 'projectId'
```

Comment:
nit: this should be `required: true` as well, even though Terraform isn't using it.

Reply:
added